core/shared/src/main/scala/matryoshka/Birecursive.scala

```diff
@@ -16,18 +16,160 @@
 
 package matryoshka
 
-import scalaz._
+import matryoshka.patterns.EnvT
+
+import scala.Predef.identity
+
+import scalaz._, Scalaz._
 
 /** A type that is both [[Recursive]] and [[Corecursive]].
-  *
-  * NB: Do not define instances of this or use it as a constraint until issue
-  *     #44 is resolved. Define an instance of [[BirecursiveT]] if possible,
-  *     otherwise define separate [[Recursive]] and [[Corecursive]] instances.
   */
-// TODO: Once this type class is actually usable, operations in `Recursive` and
-//       `Corecursive` that also have a `Corecursive.Base[T, Base]` constraint
-//       should be moved here.
-trait Birecursive[T] extends Recursive[T] with Corecursive[T]
+trait Birecursive[T] extends Recursive[T] with Corecursive[T] { self =>
+  implicit val rec: Recursive.Aux[T, Base] = self
+  implicit val corec: Corecursive.Aux[T, Base] = self
```

Comment:
Given the later `toRecursive` and `toCorecursive` definitions, do these two actually need to be implicit?